groth16-sol/src/bin/groth16-sol-utils.rs

```diff
@@ -0,0 +1,127 @@
+use ark_bn254::Bn254;
+use ark_ff::Zero;
+use circom_types::groth16::{Proof, PublicInput, VerificationKey};
+use clap::{Args, Parser, Subcommand};
+use eyre::Context;
+use std::{fs::File, path::PathBuf, process::ExitCode};
+use taceo_groth16_sol::askama::Template;
+use taceo_groth16_sol::{SolidityVerifierConfig, SolidityVerifierContext};
+
+/// Utility tools for creating and interacting with Solidity verifier contracts for BN254 Groth16 proofs. This CLI can extract a Solidity verifier from a verification key (based on the Groth16 implementation in gnark) and generate parameters for calling both the compressed and uncompressed versions of the verifier contract.
+#[derive(Debug, Parser)]
+#[command(version, about, long_about = None)]
+struct Config {
+    #[clap(subcommand)]
+    subcommand: SubCommand,
+}
+
+#[derive(Debug, Subcommand)]
+enum SubCommand {
+    GenerateCall(GenerateCallConfig),
+    ExtractVerifier(ExtractVerifierConfig),
+}
+
+#[derive(Debug, Default, Args)]
+struct GenerateCallConfig {
+    /// Path to Circom proof.
+    #[clap(long)]
+    pub proof: PathBuf,
+    /// Path to Circom public inputs.
+    #[clap(long)]
+    pub public: PathBuf,
+    /// Output of the Solidity file. Write to stdout if omitted.
+    #[clap(short, long)]
+    pub output: Option<PathBuf>,
+    /// If set, will use uncompressed points. Compressed are better when interacting with L2.
+    #[clap(long)]
+    pub uncompressed: bool,
+}
+
+#[derive(Debug, Default, Args)]
+struct ExtractVerifierConfig {
+    /// Path to Circom verification key.
+    #[clap(short, long)]
+    pub vk: PathBuf,
```

Comment:
Do we also want to be able to load ark-verification keys here? or just leave as is for now?